README.md

```diff
@@ -219,3 +219,18 @@ MaxText supports training and inference of various open models. Follow user guid
 
 * [Gemma](https://ai.google.dev/gemma): a family of open-weights Large Language Model (LLM) by [Google DeepMind](https://deepmind.google/), based on Gemini research and technology. \
 You can run decode and finetuning using instructions mentioned [here](https://github.com/google/maxtext/blob/main/getting_started/Run_Gemma.md).
+
+## Deterministic Data Input Pipeline - Grain
+
+MaxText users can optionally use [Grain](https://github.com/google/grain?tab=readme-ov-file), a deterministic data input pipeline. Grain makes the training process reproducible, disruption-proof, and debuggable. To use this pipeline:
+1. Dataset needs to be in [ArrayRecord](https://github.com/google/array_record) format, which supports random access. For converting dataset into ArrayRecord, see [instructions](https://github.com/google/array_record/tree/main/beam).
+2. ArrayRecord dataset, when hosted on GCS bucket, can only be read through [GCSFuse](https://cloud.google.com/storage/docs/gcs-fuse). The installation of GCSFuse is included in [setup.sh](https://github.com/google/maxtext/blob/main/setup.sh). However, user needs to mount the GCS bucket to a local path for each worker, using the script [setup_gcsfuse.sh](https://github.com/google/maxtext/blob/main/setup_gcsfuse.sh). As you can see, we added some default config for the mount, see [here](https://cloud.google.com/storage/docs/gcsfuse-cli) the configurable parameters of gcsfuse CLI.
```

Comment:
I removed this since it's already in setup_gcsfuse.sh (see the comments I added in this file). Because the parameters I set in the gcsfuse command are kinda arbitrary, I won't to give users the transparency in case they need to adjust